### PR TITLE
Enhance Remove DataNode Test

### PR DESCRIPTION
**Pull request by HxpSerein**
## Description

Improve remove datanode related UT/IT for complex test scenarios in region migration tests.
- Task management and conflict checking
- Running/ReadOnly nodes can be rejoined as new nodes after being removed
- Nodes and their region statuses that failed to remove are correctly rolled back

### CODE REVIEW

File: integration-test/src/test/java/org/apache/iotdb/confignode/it/removedatanode/IoTDBRemoveDataNodeITFramework.java [hunk 1]
```diff
@@ -88,7 +89,8 @@ public void successTest(
       final int configNodeNum,
       final int dataNodeNum,
       final int removeDataNodeNum,
-      final int dataRegionPerDataNode)
+      final int dataRegionPerDataNode,
+      final boolean needRestartRemoveDataNode)
```
Comment (OneSizeFitsQuorum) [inline integration-test/src/test/java/org/apache/iotdb/confignode/it/removedatanode/IoTDBRemoveDataNodeITFramework.java:93]: rejoinRemovedDataNode?

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/load/cache/node/DataNodeHeartbeatCache.java [hunk 2]
```diff
@@ -38,10 +38,6 @@ public DataNodeHeartbeatCache(int dataNodeId) {
 
   @Override
   public synchronized void updateCurrentStatistics() {
-    // The Removing status can not be updated
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/load/cache/node/DataNodeHeartbeatCache.java:41]: Why remove this line?Then the remove status maybe updated by Running or Readonly or Unknown?

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/load/cache/region/RegionCache.java [hunk 3]
```diff
@@ -65,7 +65,7 @@ public synchronized void cacheHeartbeatSample(
       return;
     }
     RegionStatus lastStatus = ((RegionHeartbeatSample) getLastSample()).getStatus();
-    if (lastStatus.equals(RegionStatus.Adding) || lastStatus.equals(RegionStatus.Removing)) {
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/load/cache/region/RegionCache.java:68]: same as above. This change may occur some bugs

File: integration-test/src/test/java/org/apache/iotdb/confignode/it/removedatanode/IoTDBRemoveDataNodeITFramework.java (hunk 1 above)
Comment (HxpSerein) [reply integration-test/src/test/java/org/apache/iotdb/confignode/it/removedatanode/IoTDBRemoveDataNodeITFramework.java]: renamed

File: iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java [hunk 4]
```diff
@@ -282,7 +282,7 @@ public void forceUpdateNodeCache(
    */
   public void removeNodeCache(int nodeId) {
     loadCache.removeNodeCache(nodeId);
-    loadCache.updateNodeStatistics();
+    loadCache.updateNodeStatistics(false);
```
Comment (OneSizeFitsQuorum) [inline iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java:285]: Why set here to false?

Comment (OneSizeFitsQuorum) [reply iotdb-core/confignode/src/main/java/org/apache/iotdb/confignode/manager/load/LoadManager.java:285]: It looks like this is also an active update, not a heartbeat update, maybe set it to true?

File: iotdb-protocol/thrift-datanode/src/main/thrift/datanode.thrift [hunk 5]
```diff
@@ -804,7 +808,7 @@ service IDataNodeRPCService {
   /**
    * Config node will stop the Data node.
    */
-  common.TSStatus stopDataNode()
+  common.TSStatus stopDataNode(TStopDataNodeReq req)
```
Comment (OneSizeFitsQuorum) [inline iotdb-protocol/thrift-datanode/src/main/thrift/datanode.thrift:811]: It appears that currently, this interface is only called when removing a datanode, and its parameter is always set to true. Can we assume this is a fixed behavior and directly proceed to delete the properties file? If, in the future, there’s a need to stop a node without deleting the properties file, we can then extend this interface. Additionally, its semantics now seem to be more than just stopping a node; perhaps it should be called stopAndClearDatanode?

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**